src/Dialogs/Task/Manager/TaskActionsPanel.cpp

```diff
@@ -103,6 +107,23 @@ TaskActionsPanel::OnDeclareClicked()
   ExternalLogger::Declare(decl, way_points.GetHome().get());
 }
 
+inline void
+TaskActionsPanel::OnDownloadClicked()
+{
+  const ComputerSettings &settings = CommonInterface::GetComputerSettings();
+  char url[256];
+  char id[20];
+
+  strcpy(id, settings.logger.pilot_weglide_id.c_str());
+  snprintf(url, sizeof(url),"https://api.weglide.org/v1/task/declaration/%s?cup=false&tsk=true",id);
+  
+  const auto cache_path = MakeLocalPath(_T("weglide"));
+  File::Delete(LocalPath(_T("weglide/weglide_declared.tsk")));
```

Comment:
If the download fails, this loses data. Doesn't sound like a good way to handle errors.